StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentMethodsView.swift

```diff
@@ -218,23 +217,31 @@ class EmbeddedPaymentMethodsView: UIView {
 
     // MARK: Mandate handling
     private func updateMandate(animated: Bool = true) {
-        self.mandateView.attributedText = mandateProvider.mandate(for: selection?.paymentMethodType,
-                                                                  savedPaymentMethod: selection?.savedPaymentMethod,
-                                                                  bottomNoticeAttributedString: nil)
-
+        let mandateText = mandateProvider.mandate(
+            for: selection?.paymentMethodType,
+            savedPaymentMethod: selection?.savedPaymentMethod,
+            bottomNoticeAttributedString: nil
+        )
+        let shouldDisplayMandate: Bool = {
+            guard let mandateText else {
+                return false
+            }
+            return shouldShowMandate && !mandateText.string.isEmpty
+        }()
+        mandateView.attributedText = mandateText
+        let updateMandateUI = {
+            let spacing = shouldDisplayMandate ? 12.0 : 0
+            let mandateViewIndex = self.stackView.arrangedSubviews.firstIndex(of: self.mandateView)! - 1
```

Comment:
1. Rather than crashing if the mandate (and the sep isn't there, maybe we should bail, and send an event?

2. nit: the name `mandateViewIndex` isn't that accurate since we are subtracting one from the index.

Reply:
event is overkill IMO, our tests will exercise this code well, but I'll make this not crash in production